Remove release-cycle pre-1.0 note

korthout opened the pull request:
Considering Zeebe release `1.0.0` is nearing, I assume we should remove the note about breaking changes before Zeebe 1.0

akeller commented:
@korthout can you take a look at this PR? I'm advocating we move this release-cycle content more up-level to handle all Camunda Cloud components. I am having trouble figuring out how we want to describe things pre-1.0. Maybe you have inputs there?

korthout commented:
@akeller Ah good point. I was unaware of that change. I just saw this page and thought let's improve it 😄 

To be honest, the more I think about it. Perhaps it makes sense to leave the comment up while we still support pre-1.0 minor versions. Consider a strange bug that requires us to make breaking changes in order to fix it, then we might have a 0.26.x or a 0.27 release to fix a bug with actual breaking changes. I don't imagine that this will actually happen though. 

@npepinpe Do you have any opinions on this?

npepinpe commented:
We will keep supporting 0.26.x until 1.1 - we can remove the note at that point. But I don't think it's a big deal to be honest if we remove it now. I'm happy to defer to our product team.

pihme commented:
@akeller I removed myself as reviewer and added you, instead, so you can coordinate this change with #210.  

In my personal opinion, I would remove this note in the 1.0.0 content. We do have versioned documentation now and in the older versions this can stay at the current position and with the current content. 
But I don't see a need to talk about pre 1.0.0 version in the 1.0.0 documentation.

And for 1.0.0 aside from removing the note it might make sense to talk about public APIs and stability that customers can expect. But this is probably more a question to product management and legal department, what we want to communicate here.

korthout commented:
I agree with this one:
> I don't see a need to talk about pre 1.0.0 version in the 1.0.0 documentation.

@akeller I've rebased the changes onto `structural-GA-changes` and changed the target for this PR.
